src/__test__/components/data-management/SamplesTable.test.jsx

```diff
@@ -103,45 +108,47 @@ describe('Samples table', () => {
   });
 
   it('Should not show option to download datasets if samples are available', async () => {
-    await act(async () => {
-      render(
-        <Provider store={storeState}>
-          {samplesTableFactory()}
-        </Provider>,
-      );
-    });
+    await renderSamplesTable(storeState);
 
     expect(screen.queryByText(/Start uploading your samples by clicking on Add samples./i)).toBeNull();
     expect(screen.queryByText(/Don't have data\? Get started using one of our example datasets:/i)).toBeNull();
   });
 
   it('Should show all the samples', async () => {
-    await act(async () => {
-      render(
-        <Provider store={storeState}>
-          {samplesTableFactory()}
-        </Provider>,
-      );
-    });
+    await renderSamplesTable(storeState);
 
     Object.values(samples).forEach((sample) => {
       expect(screen.getByText(sample.name)).toBeInTheDocument();
     });
   });
 
+  it('Should show an error if a sample fails to upload', async () => {
+    const missingSampleState = _.cloneDeep(storeState.getState());
+    const createMockStore = configureMockStore([thunk]);
```

Comment:
I had to use the mock store to replicate the condition where the sample id exists in `projects` but not `samples` as there is no easy way to replicate that state using the real sore.